docs/tutorials/04-design-automation/_shared/nodejs-vscode/setup/workitem.mdx

```diff
@@ -7,42 +7,37 @@ This is where we actually start the Design Automation. This endpoint also upload
 ```js
 /// <summary>
 /// Direct To S3
+/// ref : https://aps.autodesk.com/blog/new-feature-support-direct-s3-migration-inputoutput-files-design-automation
 /// </summary>
-const prepareInputUrl = async (
-  bucketKey,
-  objectKey,
-  opts,
-  oAuthClient,
-  oAuthToken
-) => {
-  let objectS3Download = await new ForgeAPI.ObjectsApi().getS3DownloadURL(
-    bucketKey,
-    objectKey,
-    opts,
-    oAuthClient,
-    oAuthToken
-  );
-  return objectS3Download.body.url;
-};
-const prepareOutputUrl = async (
-  bucketKey,
-  objectKey,
-  opts,
-  oAuthClient,
-  oAuthToken
-) => {
-  let objectS3Upload = await new ForgeAPI.ObjectsApi().getS3UploadURL(
-    bucketKey,
-    objectKey,
-    opts,
-    oAuthClient,
-    oAuthToken
-  );
-  return {
-    outputUrl: objectS3Upload.body.urls[0],
-    uploadKey: objectS3Upload.body.uploadKey,
-  };
+
+const getObjectId = async (bucketKey, objectKey, req) => {
+  try {
+    let contentStream = _fs.createReadStream(req.file.path);
+    let uploadResponse = await new ForgeAPI.ObjectsApi().uploadResources(
+      bucketKey,
+      {
+        objectKey: objectKey,
+        data: contentStream,
+        length: req.file.size,
+      },
+      {
+        useAcceleration: false,
+        minutesExpiration: 20,
+        onUploadProgress: (data) => console.warn(data),
+      },
+      req.oauth_client,
+      req.oauth_token
+    );
+    console.log(uploadResponse[0].completed.objectId);
+    return uploadResponse[0].completed.objectId;
+  } catch (ex) {
+    console.error(ex);
+    return res.status(500).json({
```

Comment:
The `res` variable is probably not defined here, right?

Reply:
I think you'll need to re-throw the exception out of this method, otherwise the exception would be "consumed" and the calling code would actually create a work item as if nothing happened.